swagger.yaml

```diff
@@ -2510,6 +2566,19 @@ definitions:
         type: string
         description: Json formatted string containing the user overrides for the host's pointer ignition
         example: '{"ignition": {"version": "3.1.0"}, "storage": {"files": [{"path": "/tmp/example", "contents": {"source": "data:text/plain;base64,aGVscGltdHJhcHBlZGluYXN3YWdnZXJzcGVj"}}]}}'
+      installer_args:
+        type: string
+
+
+  host-update-params:
```

Comment:
if looks like a generic API, so maybe the parameters should be updated the same as in cluster update.
all the parameters should be nullable so if someone will add more parameters there will be an option to update only some parameters and an option to remove parameters with empty string (or list in this case)

Reply:
I think I'm unclear on what exactly you're referencing here. Do you want me to declare (and implement) a full host update? Or are you talking specifically about the installer parameters field?

I went with a straight override of the array for a few reasons:
1. It works the same way as any other property of a host. It would be strange to merge some, but overwrite others in a generic PATCH endpoint on the host resource.
2. Doing a delete in an array merge is hard. As you said we'd have to implement some additional values here to indicate what needs to be added, and what needs to be removed.
3. It works the same way as the other customizations. We don't merge subsequent calls to the ignition or install config overrides into earlier ones. They all overwrite.

Reply:
There are two issues that i see here.
1. We update hosts in two places, in cluster patch api and here - @avishayt we need ether split the api or put it under cluster patch.
2. if the parameter is not nullable then if anyone will add another parameter he will create a bug, take a look at how cluster update is implemented. We don't set all the parameters every time, we enable all the parameters be nullable, this way if use don't set some parameter we don't overwrite it with the default value, so the difference is between enabling the user to delete existing value and the option not to set all the values when calling patch API.

Reply:
:+1: okay so when you say "parameter" here you're talking about the entire `installer_args` array, not the individual elements (which are command line "parameters"). All my points were assuming the opposite :sweat_smile: 

Yes, I agree that adding a new host endpoint for this is confusing given that we already update hosts in the cluster endpoint. I would prefer all host updates happen on the hosts endpoint.

I can definitely make `installer_args` nullable here, do you also think I should add the two other fields? It should be a pretty easy refactoring and won't break the existing cluster API (we can just leave that for now). Alternatively, I could leave this with the one nullable field or add the others and ignore them on the backend. What do you think?